\documentclass[12pt,oneside]{article}
\usepackage[margin=2cm]{geometry}
\usepackage{multirow}
\usepackage{graphicx}
\usepackage[colorlinks=true,citecolor=black,urlcolor=blue]{hyperref}

% Use the boxed version of the \embed command to guide trimming:
% trim = left, bottom, right, top
%\newcommand{\embed}[5]{\begin{center}\fbox{\includegraphics[trim=#1 #2 #3 #4,clip,width=\textwidth]{#5}}\end{center}\clearpage}
\newcommand{\embed}[5]{\begin{center}\includegraphics[trim=#1 #2 #3 #4,clip,width=\textwidth]{#5}\end{center}\clearpage}

\newcommand{\talkposter}[1]{\centerline{\small\noindent This talk is accompanied by poster \##1.}}
\newcommand{\onlyposter}[1]{\centerline{\small\noindent Poster \##1.}}

\usepackage{fancyhdr}
\setlength\headheight{2cm}
\fancyhead{}
\fancyfoot{}
\renewcommand{\headrulewidth}{1pt}
\renewcommand{\footrulewidth}{0pt}
\lhead{Bioinformatics Open Source Conference (BOSC) 2015\\Boston}
\rhead{\includegraphics[height=1cm]{../../logos/100px-Pear.png}}

\newcommand{\session}[3]{\clearpage\lhead{BOSC 2015 -- \textit{#1}\\#2\\#3}\newpage}

\title{%Hack to get the logo on the PDF front page:
$16^{th}$ Annual \\
Bioinformatics Open Source Conference\\
BOSC 2015\\
~\\
\includegraphics[width=5cm]{../../logos/bosc-pear.pdf}\\
%Hack to get some white space using a blank line:
~\\
\url{http://www.open-bio.org/wiki/BOSC_2015}
}

\date{10 \& 11 July 2015, Dublin, Ireland}

\begin{document}
\pagestyle{fancy}

%\maketitle

%\thispagestyle{empty}

% trim = left, bottom, right, top

\session{Data Science}{Day One, 10 July 2015, 10:45 -- 12:30}{Session chair: Rob Davey}

% Submission 24 - Apollo: Scalable & collaborative curation for improved comparative genomics 
\talkposter{1}
\embed{2.5cm}{3cm}{1.25cm}{2.6cm}{24-Apollo.pdf}
% Submission 19 - GOexpress: A R/Bioconductor package for the identification and visualisation of robust gene ontology signatures through supervised learning of gene expression data
\talkposter{2}
\embed{2.5cm}{3cm}{1.25cm}{2.6cm}{19-GOexpress.pdf}
% Submission 49 - Arvados: A Free Software Platform for Big Data Science
\embed{2cm}{4cm}{2cm}{2.5cm}{49-arvados.pdf}
% Submission 54 - Bringing Hadoop into Bioinformatics with Cloudgene and CloudMan
\embed{2.5cm}{3cm}{1.25cm}{2.5cm}{54-cloudgene.pdf}
% Submission 23 - Segway: semi-automated genome annotation
\talkposter{3}
\embed{2.5cm}{7cm}{2.5cm}{2.5cm}{23-Segway.pdf}
% Submission 27 - QualiMap 2.0: quality control of high throughput sequencing data
\talkposter{4}
\embed{2.5cm}{7cm}{2.5cm}{2.5cm}{27-qualimap.pdf}
% Submission 26 - A Genomics Virtual Laboratory
\talkposter{5}
\embed{2.5cm}{6cm}{1.75cm}{2.5cm}{26-cloudman.pdf}
% Submission 52 - BioSolr: Building better search for bioinformatics
\talkposter{6}
\embed{2.5cm}{7cm}{2.5cm}{2.5cm}{52-BioSolr.pdf}
% Submission 35 - Prioritization of structural variants based on known biological information
\embed{2.5cm}{5cm}{2.5cm}{2.5cm}{35-bcbio.pdf}

\session{Standards and Interoperability}{Day One, 10 July 2015, 14:00 -- 15:30}{Session chair: Peter Cock}

% Submission 30 - Portable workflow and tool descriptions with the CWL
\talkposter{7}
\embed{2.5cm}{6cm}{2.5cm}{2.5cm}{30-workflow.pdf}
% Submission 46 - From peer-reviewed to peer-reproduced: a role for research objects in scholarly publishing in the life sciences
\embed{2.5cm}{3.5cm}{1.5cm}{2.8cm}{46-case-study.pdf}
% Submission 42 - Demystifying the Interoperability of Disparate Genomic Resources
\embed{2.5cm}{3cm}{1cm}{2.5cm}{42-demyst-galaxy.pdf}
% Submission 32 - Increasing the utility of Galaxy workflows
\embed{2.5cm}{2.75cm}{1cm}{2.5cm}{32-galaxy-workflows.pdf}
% Submission 22 - Kipper: A software package for sequence database versioning for Galaxy bioinformatics servers
\embed{2.5cm}{2.75cm}{1cm}{2.5cm}{22-kipper.pdf}
% Submission 37 - Evolution of the Galaxy tool ecosystem - happier developers, happier users
\talkposter{8}
\embed{2.5cm}{3cm}{0.75cm}{2cm}{37-galaxy.pdf}
% Submission 45 - Bionode - Modular and universal bioinformatics
\talkposter{9}
\embed{2.5cm}{7cm}{1.75cm}{2.75cm}{45-bionode.pdf}
% Submission 48 - The EDAM Ontology
\talkposter{10}
\embed{2.5cm}{12cm}{2.5cm}{2.75cm}{48-EDAM.pdf}

\session{Open Science and Reproducibility}{Day Two, 10 July 2015, 10:45 -- 12:30}{Session chair: M{\'o}nica Mu{\~n}oz-Torres }

% Submission 47 - A curriculum for teaching Reproducible Computational Science bootcamps
\embed{2.5cm}{10cm}{1cm}{2.75cm}{47-repo-sci-curriculum.pdf}
% Submission 9 - Research shared: www.researchobject.org
\talkposter{11}
\embed{2.5cm}{6cm}{2.5cm}{3.25cm}{9-researchobject.pdf}
% Submission 3 - Nextflow: a tool for deploying reproducible computational pipelines
\talkposter{12}
\embed{2.5cm}{2.5cm}{1cm}{2.6cm}{3-nextflow.pdf}
% Submission 41 - Free beer today: how iPlant + Agave + Docker are changing our assumptions about reproducible science
\talkposter{13}
\embed{2.5cm}{8cm}{2.5cm}{2.5cm}{41-free-beer.pdf}
% Submission 36 - The 500 builds of 300 applications in the HeLmod repository will at least get you started on a full suite of scientific applications
\embed{2.5cm}{6cm}{2cm}{2.5cm}{36-helmod.pdf}
% Submission 39 - Bioboxes: Standardised bioinformatics tools using Docker containers.
\talkposter{14}
\embed{2.5cm}{3.25cm}{0cm}{1.25cm}{39-bioboxes.pdf}
% Submission 43 - The perfect fit for reproducible interactive research: Galaxy, Docker, IPython
\embed{2.5cm}{6cm}{2.5cm}{2.75cm}{43-galaxy-docker.pdf}
% Submission 28 - COPO: Bridging the Gap from Data to Publication in Plant Science
\talkposter{15}
\embed{2.5cm}{3.5cm}{1.5cm}{2.75cm}{28-COPO.pdf}
% Submission 15 - ELIXIR UK building on Data and Software Carpentry to address the challenges in computational training for life scientists
\talkposter{16}
\embed{2.5cm}{2.5cm}{1cm}{2.65cm}{15-ELIXIR.pdf}
% Submission 33 - Parallel recipes: towards a common coordination language for scientific workflow management systems
\talkposter{17}
\embed{2.5cm}{2.5cm}{1cm}{2.65cm}{33-precipes.pdf}
% Submission 1 - openSNP - personal genomics and the public domain
\talkposter{18}
\embed{2.5cm}{3cm}{1.25cm}{2.65cm}{1-openSNP.pdf}

\session{Translational Bioinformatics}{Day Two, 11 July 2015, 14:00 -- 14:40}{Session chair: Brad Chapman}

% Submission 4 - CIViC: Crowdsourcing the Clinical Interpretation of Variants in Cancer
\talkposter{19}
\embed{2.5cm}{3.25cm}{1.5cm}{2.65cm}{4-civic.pdf}
% Submission 7 - From Fastq To Drug Recommendation - Automated Cancer Report Generation using OncoRep \&amp; Omics Pipe
\talkposter{20}
\embed{2.5cm}{3.25cm}{1.5cm}{2.65cm}{7-OncoRep.pdf}
% Submission 25 - Cancer Informatics Collaboration and Computation: Two Initiatives of the U.S. National Cancer Institute
\talkposter{21}
\embed{2.5cm}{4cm}{2cm}{3cm}{25-nciphub.pdf}

\session{Bioinformatics Open Source Project Updates}{Day Two, 11 July 2015, 14:40 -- 15:30}{Session chair: Nomi Harris}

% Submission 17 - Biopython Project Update 2015
\embed{2cm}{1.5cm}{0cm}{1cm}{17-Biopython.pdf}
% Submission 21 - The biogems community: Challenges in distributed software development in bioinformatics
\embed{3.55cm}{4.5cm}{1cm}{4.5cm}{21-biogems.pdf}
% Submission 16 - Apache Taverna: Sustaining research software at the Apache Software Foundation
\talkposter{22}
\embed{2.5cm}{3.25cm}{1.5cm}{2.75cm}{16-Taverna.pdf}

\session{Visualization}{Day Two, 11 July 2015, 16:00 -- 16:30}{Session chair: Karsten Hokamp}

% Submission 18 - Simple, Shareable, Online RNA Secondary Structure Diagrams
\embed{2.5cm}{4cm}{2cm}{2.75cm}{18-forna.pdf}
% Submission 34 - BioJS 2.0: an open source standard for biological visualization
\talkposter{23}
\embed{2.5cm}{2.5cm}{1cm}{2.65cm}{34-BioJS.pdf}
% Submission 10 - Visualising Open PHACTS linked data with widgets
\talkposter{24}
\embed{2.5cm}{5cm}{2.25cm}{2.65cm}{10-PHACTS.pdf}

\session{Late-Breaking Lightning Talks}{Day Two, 10 July 2015, 16:30 -- 17:00}{Session chair: Hilmar Lapp}

% Submission 12 - Biospectra-by-sequencing genetic analysis platform
\talkposter{25}
\embed{2.5cm}{5cm}{2.5cm}{2.5cm}{12-Biospectra.pdf}
% Submission 67 - PhyloToAST: Bioinformatics tools for species-level analysis and visualization of complex microbial communities
\embed{2.5cm}{3.6cm}{2cm}{2.85cm}{67-phylotoast.pdf}
% Submission 57 - Otter/ZMap/SeqTools: A productive alternative to web browser genome visualisation
\talkposter{26}
\embed{2.5cm}{4.25cm}{2.25cm}{2.85cm}{57-annosoft.pdf}
% Submission 65 - aRchive: enabling reproducibility of Bioconductor package versions
\talkposter{27}
\embed{2.5cm}{8cm}{2.5cm}{2.5cm}{65-aRchive.pdf}
% Submission 58 - Developing an Arvados BWA-GATK pipeline
\talkposter{28}
\embed{2.5cm}{2cm}{1.5cm}{3.75cm}{58-arvados.pdf}
% Submission 63 - Out of the box cloud solution for Next-Generation Sequencing analysis
\talkposter{29}
\embed{2.5cm}{2.5cm}{1cm}{2.65cm}{63-molgenis.pdf}

\session{Poster-only Abstracts}{Days One \& Two, 10 \& 11 July 2015, 13:00 -- 14:00}{}

%Have 29 doing talk and poster

\onlyposter{30}
% Submission 66 - Aequatus: Visualising complex similarity relationships among species
\embed{2.5cm}{3.5cm}{0.75cm}{2.6cm}{66-aequatus.pdf}

\onlyposter{31}
% Submission 6 - MOLGENIS Workbench for Systems Medicine
\embed{2.5cm}{4cm}{1.5cm}{2.75cm}{6-molgenis.pdf}

\onlyposter{32}
% Submission 13 - SPINGO: a rapid species-classifier for microbial amplicon sequences
\embed{2.5cm}{12cm}{2.25cm}{2.65cm}{13-SPINGO.pdf}

\onlyposter{33}
% Submission 51 - ANNOgesic - A computational pipeline for RNA-Seq based transcriptome annotations of bacteria
\embed{2.5cm}{12cm}{1.8cm}{2.85cm}{51-ANNOgesic.pdf}

\onlyposter{34}
% Submission 44 - BioXSD - a data model for sequences, alignments, features, measured and inferred values
\embed{2.5cm}{3.25cm}{1cm}{2.8cm}{44-bioxsd.pdf}

\onlyposter{35}
% Submission 60 - MGkit: A Metagenomic Framework For The Study Of Microbial Communities
% (replaced submission 20)
\embed{2.5cm}{12cm}{2.25cm}{2.85cm}{60-mgkit.pdf}

\onlyposter{36}
% Submission 64 - From scaffold to submission in a day: a new software pipeline for rapid genome annotation and analysis
\embed{2.5cm}{2.5cm}{0.25cm}{2.65cm}{64-annot-nf.pdf}

% trim = left, bottom, right, top

\end{document}
